File: versioned_docs/version-1.11.x/examples/rs/examples/data_ingestion_batch.rs
use bytes::Bytes;
use reduct_rs::{RecordBuilder, ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a client instance, then get or create a bucket
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();
    let bucket = client.create_bucket("test").exist_ok(true).send().await?;

    // Prepare a batch of records
    let record_1 = RecordBuilder::new()
        .timestamp_us(1_000_000)
        .data("Records #1")
        .build();

    let record_2 = RecordBuilder::new()
        .timestamp_us(2_000_000)
        .data("Records #2")
        .build();

    // Send the batch of records to the "rs-example" entry
    let errors = bucket
        .write_batch("rs-example")
        .add_records(vec![record_1, record_2])
        .send()
        .await?;

    // Check if there are any errors and return first one
    for (_timestamp, error) in errors {
        return Err(error);
    }

    Ok(())
}

File: versioned_docs/version-1.17.x/examples/py/src/data_ingestion_batch.py
import time
import asyncio
from typing import Dict

from reduct import Client, Bucket, Batch, ReductError


async def main():
    # Create a client instance, then get or create a bucket
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        bucket: Bucket = await client.create_bucket("my-bucket", exist_ok=True)

        # Prepare a batch of records
        batch = Batch()
        batch.add(
            "2024-02-02T10:00:00",
            b"Records #1",
        )
        batch.add(
            "2024-02-02T10:00:01",
            b"Records #2",
        )
        batch.add(
            "2024-02-02T10:00:02",
            b"Records #3",
        )

        # Write the batch to the "py-example" entry of the bucket
        errors: Dict[int, ReductError] = await bucket.write_batch("py-example", batch)

        # Check statuses and raise first error
        for timestamp, err in errors.items():
            raise err


loop = asyncio.get_event_loop()
loop.run_until_complete(main())
